### PR TITLE
Add More IO80211Family headers and fix structs

### PR DESCRIPTION
**Pull request by zxystd**
- Add 10.13 and 10.14 `IO80211Family` headers support.
- Add `IO80211VirtualInterface` and `IO80211P2PInterface`, these are for enabling `AirDrop` and `Handoff` etc.
- fix `apple80211_assoc_data` `apple80211_key` `apple80211_capability_data` structure declaration.
- add `apple80211_tx_nss_data` `apple80211_awdl_peer_traffic_registration` `apple80211_awdl_election_metric` `apple80211_awdl_sync_enabled` `apple80211_awdl_sync_frame_template` `apple80211_virt_if_delete_data` `apple80211_virt_if_create_data` `apple80211_ht_capability`.

### CODE REVIEW

File: Headers/IOKit/80211/IO80211Controller.h [hunk 1]
```diff
@@ -127,255 +127,197 @@ struct apple80211_lteCoex_report;
 typedef IOReturn (*IOCTL_FUNC)(IO80211Controller*, IO80211Interface*, IO80211VirtualInterface*, apple80211req*, bool);
 extern IOCTL_FUNC gGetHandlerTable[];
 extern IOCTL_FUNC gSetHandlerTable[];
+#define __int64 int
```
Comment (vit9696) [inline Headers/IOKit/80211/IO80211Controller.h]: These are hacks from IDA, better to replace them with C types to be honest.

File: Headers/IOKit/80211/IO80211Controller.h [hunk 2]
```diff
@@ -127,255 +127,197 @@ struct apple80211_lteCoex_report;
...
-    void configureAntennae(void);
+    
+//public:
+//    IO80211SkywalkInterface* getInfraInterface(void);
+//    IO80211ScanManager* getPrimaryInterfaceScanManager(void);
+//    IO80211ControllerMonitor* getInterfaceMonitor(void);
+//    IOReturn addReporterLegend(IOService *,IOReporter *,char const*,char const*);
+//    IOReturn removeReporterFromLegend(IOService *,IOReporter *,char const*,char const*);
+//    IOReturn unlockIOReporterLegend(void);
+//    void lockIOReporterLegend(void);//怀疑对象，之前是返回int
```
Comment (vit9696) [inline Headers/IOKit/80211/IO80211Controller.h]: Why are these commented out? Also, can you replace Chinese with English here and below?

**Comment by vit9696** [comment]
There also are quite a lot of spacing changes. Could we avoid these?

File: Headers/IOKit/80211/IO80211Controller.h (hunk 1 above)
Comment (zxystd) [reply Headers/IOKit/80211/IO80211Controller.h]: ok.

**Comment by zxystd** [comment]
> There also are quite a lot of spacing changes. Could we avoid these?

How can I do? is there a format rule?

File: Headers/IOKit/80211/IO80211Controller.h (hunk 2 above)
Comment (zxystd) [reply Headers/IOKit/80211/IO80211Controller.h]: non-virtual functions affects the link to the IO80211Family, and it is different from various versions of macOS, so I commented then now.
Yes, the comment I will change to English.

**Comment by vit9696** [comment]
Unsure if there is a way other than using something like Araxis Merge. It is not like we have clang-format here or anything though we could.

**Comment by zxystd** [comment]
> Unsure if there is a way other than using something like Araxis Merge. It is not like we have clang-format here or anything though we could.

Ok, I will revert them to your space format.

File: Headers/IOKit/80211/IO80211Controller.h (hunk 2 above)
Comment (vit9696) [reply Headers/IOKit/80211/IO80211Controller.h]: Without calling these functions they will not link automatically. Could leave uncommented really.

File: Headers/IOKit/80211/apple80211_ioctl.h [hunk 3]
```diff
@@ -34,7 +34,7 @@
 #error "Please define __IO80211_TARGET to the requested version"
 #endif
 
-#include <IOKit/80211/apple80211_var.h>
+#include "apple80211_var.h"
```
Comment (vit9696) [inline Headers/IOKit/80211/apple80211_ioctl.h]: Should not be local include

**Pull request merged**